server/processors/src/main/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriver.java

```diff
@@ -164,4 +158,44 @@ protected void deriveNodeDetails(Trace trace, List<Node> nodes, List<NodeDetails
         commonProperties.addAll(trace.getProperties(Constants.PROP_PRINCIPAL));
         return commonProperties;
     }
+
+    /**
+     * This method calculates the elapsed time associated with the child nodes of the
+     * supplied parent node. If the supplied node is not a parent, or if the child nodes are
+     * performed concurrently and therefore not impacting the time spent in the parent node
+     * then this method will return an elapsed time of 0.
+     *
+     * @param n The parent node
+     * @return The elapsed time of the child nodes
+     */
+    protected long calculateChildElapsedTime(Node n) {
+        long childElapsed = 0;
+        if (n.containerNode()) {
+            long startTime=n.getTimestamp() + n.getDuration();
```

Comment:
They are being processed independently, and unless there is something wrong with the data provided in the child nodes,then after processing the child nodes the start time will be less than the end time.